portfolio/src/main/data_proccessing/processor.py

```
@@ -7,31 +7,44 @@
-stored within a database."""
+# processor.py is used to clean csv's and create data suitable to be
+# stored within a database.
 
 import pandas as pd
-import numpy as np
 
-# TODO(kofimeighan): Create a function that can iterate over all files within 
-# the raw_data directory
-RAW_DATA_FILEPATH = 'raw_data/police_killings_2013.csv'
+RAW_TIME_FILEPATH = 'raw_data/police_killings_2013.csv'
 CLEAN_DATA_FILEPATH = 'clean_data/police_killings_2013.csv'
-COLUMNS_OF_INTEREST = ['name', 'race', 'date', 'address', 'city',
-                       'state', 'zipcode',]
-dataframe = pd.read_csv(RAW_DATA_FILEPATH, index_col=0)
-csv_columns = dataframe.columns
-columns_to_keep = []
-
-# TODO(kofimeighan): save the zipcode as an int not a long within the database
-for csv_column in csv_columns:
-    stripped_csv_column = csv_column.lower().replace("'", "")
-    for column_of_interest in COLUMNS_OF_INTEREST:
-        if column_of_interest.lower() in stripped_csv_column:
-            columns_to_keep.append(csv_column)
-            break
-
-dataframe = dataframe[columns_to_keep]    
-dataframe.to_csv(CLEAN_DATA_FILEPATH, index=False)
+TIME_COLUMNS = ['date', 'name']
+KIND = "Task"
+TABLE_NAME = "Police Violence Data"
+PROJECT_ID = "silenceisviolence"
+
+
+def new_frame(file_path, column_names):
+  '''
+  args: (path to csv files, names of columns to retain)
+  rets: pd.Dataframe containing desired columns from csv 
+  '''
+  tempframe = pd.read_csv(file_path)
+
+  columns_to_keep = []
+  for column in tempframe:
+    for column_of_interest in column_names:
+      if column_of_interest in column.lower():
+        columns_to_keep.append(column)
+        break
+
+  tempframe = tempframe[columns_to_keep]
+  return tempframe
+
+
+def timSeries():
+  timeframe = new_frame(RAW_TIME_FILEPATH, TIME_COLUMNS)
+  timeframe.columns = ['Totals', 'Date']
+  timeframe['Date'] = pd.to_datetime(timeframe['Date'], format='%m/%d/%y',
+                                     errors='ignore')
```

Why are we ignoring errors instead of fixing them or dropping rows?

---

Ignore seems to drop the unwanted rows. I tried 'raise’ and ‘coerce’ but those did not work as expected.